src/platform/K32W/KeyValueStoreManagerImpl.cpp

```diff
@@ -39,61 +42,89 @@ namespace PersistedStorage {
 
 KeyValueStoreManagerImpl KeyValueStoreManagerImpl::sInstance;
 
+/* hashmap having:
+ * 	- the matter key value as key;
+ * 	- internal PDM identifier as value;
+ */
+std::unordered_map<std::string, uint8_t> kvs;
```

Comment:
We should avoid using stl containers which allocate memory in heap.

Reply:
What is the RAM availability of K32W?

Generally I would be ok with stl on large platforms (e.g. linux or darwin) in platform code. However here I agree that an embedded device likely should not use STL that may allocate heap.

Reply:
Also separate code review comments:

name of 'kvs' does not seem to indicate global. use gKvsKeyIdMap or something similar.
Where is this variable persisted? If device reboots,  how is this map and list below initialized?

Reply:
That is not a problem if it's in platform code. They use STL in lots of other things already on K32W

Reply:
@andy31415 , K32W has 152K of SRAM. Let's keep stl for the moment and revise it later when the KVS API will change (maybe strings won't be used as keys). I also refactored the names of the global variables.